{% extends "base.html" %}
{% block content %}
<h1>Specifications followed by this database</h1>

<h2>The RDAT file format</h2>
<p>The RNA Data (RDAT) text file format is an annotation-based specification for RNA structure mapping experiments that is both computer
friendly and can be easily read by humans. The RMDB database schema closely follows the
organization of RDAT files, making it easier to manage and expose each entryâ€™s data.</p>
<p>RDAT files have a hierarchical structure and are comprised by three main sections: the general, outer section, the construct section, and the data section. An RDAT file can contain multiple construct sections, each describing a specific construct (for example, an RNA molecule of interest). Each construct section contains multiple data sections, which describe a particular electropherogram trace or set of read counts.
<p>RDAT files depend heavily on annotations, as each section can have multiple annotations that further describe the experiments and the data contained (for example, what chemical modifiers have been used, the construct concentrations, if the data has been background substracted or not, etc). These annotations are also hierarchical, in a sense that a parent annotation is inherited to all constructs (for example, modifier:CMCT given as an annotation for the general section would indicate that all constructs have been modified by CMCT). This hierarchy can be overridden by more section specific annotations (in the example above, if there is a construct that had no modification, we place an annotation modifier:None to its construct section). Whole attributes that can appear in multiple sections, such as SEQPOS, are also hierarchical and can be used to specify attributes for all the data contained in the file to avoid redundancy. Attributes of each section, as well as annotation syntax and semantics are described throughout this document.</p>

<p>The following is an example of a small RDAT file for single replicate experiments with the data for the medloop construct:</p>
<pre>
RDAT_VERSION	0.32
NAME	MedLoop
SEQUENCE	GGAACGACGAACCGAAAACCGAAGAAAUGAAGAAAGGUUUUCGGUACCGACCUGAAAACCAAAGAAACAACAACAACAAC
STRUCTURE	..........((((((((((...............))))))))))...................................
OFFSET	-10
SEQPOS  50 49 48 47 46 45 44 43 42 41 40 39 38 37 36 35 34 33 32 31 30 29 28 27 26 25 24 23 22 21 20 19 18 17 16 15 14 13 12 11 10  9  8  7  6  5  4  3  2  1  0 -1 -2 -3 -4 -5 -6 -7 -8 -9
SEQPOS	G-9	G-8	A-7	A-6	C-5	G-4	A-3	C-2	G-1	A0	A1	C2	C3	G4	A5	A6	A7	A8	C9	C10	G11	A12	A13	G14	A15	A16	A17	U18	G19	A20	A21	G22	A23	A24	A25	G26	G27	U28	U29	U30	U31	C32	G33	G34	U35	A36	C37	C38	G39	A40	C41	C42	U43	G44	A45	A46	A47	A48	C49	C50
MUTPOS	WT	WT
ANNOTATION	experimentType:StandardState	chemical:Na-HEPES:50mM(pH8.0)	temperature:24C

COMMENT	Example RDAT file with all fields filled: DMS and CMCT data for the MedLoop RNA.

ANNOTATION_DATA:1	modifier:DMS
ANNOTATION_DATA:2	modifier:CMCT

REACTIVITY:1        161.1038   70.2383   75.5198   88.3231   59.1168   44.1307   14.3177   15.7294   48.0327   27.8253   58.5714   22.2025   28.4068    9.0458   51.8737    6.6602   10.6241    0.0000  123.8299    5.9402    2.5001    3.3970    2.4758   10.8160    0.0000   60.2178   79.3310   72.5287    0.0362   57.8198   60.5596    7.6400    3.2410   62.5949   47.8698   43.7576    0.9151   49.9397   50.9380    6.8794   18.1433    7.1330   11.3588    5.7703    4.4164    6.8753    4.7442   24.0865    7.5582    8.6075   22.9732    2.5496    9.5329   26.6948    5.3731   11.8460   27.7534   44.8170  148.2240  169.3202
REACTIVITY:2        326.7871   50.8572   13.5739   24.2813    7.4603    2.3846   45.6323  263.8077   38.4869   12.8314    1.4421   17.6450   26.5011    3.1060    2.4043   12.1818    5.0894    0.0000   38.8606    3.0631    3.5447    2.5682    2.9326   11.2679    0.3434    7.9297   10.2588    5.6163   51.8502   21.8487   20.5006  118.0059  278.5901   24.2993    8.9043    3.9261   54.7244   10.2562    8.7679   16.9593   24.8004    4.1159    5.4285    1.4056    0.9583    2.8067    5.2360   24.2889    5.8440    3.3078    4.0801   15.6673   16.6094    6.0948   18.4954    7.0075    2.6482  183.4449  308.4945  177.4841

REACTIVITY_ERROR:1        34.2208   16.0477   17.1040   19.6646   13.8234   10.8261    4.8635    5.1459   11.6065    7.5651   13.7143    6.4405    7.6814    3.8092   12.3747    3.3320    4.1248    2.0000   26.7660    3.1880    2.5000    2.6794    2.4952    4.1632    2.0000   14.0436   17.8662   16.5057    2.0072   13.5640   14.1119    3.5280    2.6482   14.5190   11.5740   10.7515    2.1830   11.9879   12.1876    3.3759    5.6287    3.4266    4.2718    3.1541    2.8833    3.3751    2.9488    6.8173    3.5116    3.7215    6.5946    2.5099    3.9066    7.3390    3.0746    4.3692    7.5507   10.9634   31.6448   35.8640
REACTIVITY_ERROR:2        67.3574   12.1714    4.7148    6.8563    3.4921    2.4769   11.1265   54.7615    9.6974    4.5663    2.2884    5.5290    7.3002    2.6212    2.4809    4.4364    3.0179    2.0000    9.7721    2.6126    2.7089    2.5136    2.5865    4.2536    2.0687    3.5859    4.0518    3.1233   12.3700    6.3697    6.1001   25.6012   57.7180    6.8599    3.7809    2.7852   12.9449    4.0512    3.7536    5.3919    6.9601    2.8232    3.0857    2.2811    2.1917    2.5613    3.0472    6.8578    3.1688    2.6616    2.8160    5.1335    5.3219    3.2190    5.6991    3.4015    2.5296   38.6890   63.6989   37.4968

XSEL 93.00   126.00   158.00   187.00   214.00   238.00   260.00   279.00   299.00   327.00   357.00   382.00   405.00   432.00   460.00   487.00   513.00   523.61   541.00   568.00   593.00   616.00   637.00   656.00   675.00   697.00   724.00   751.00   775.00   799.00   825.00   850.00   871.00   895.00   921.00   947.00   970.00   992.00  1017.00  1042.00  1066.00  1093.00  1121.00  1148.00  1173.00  1197.00  1220.00  1242.00  1268.00  1294.00  1320.00  1343.00  1365.00  1389.00  1412.00  1435.00  1461.00  1487.00  1519.00  1539.00

XSEL_REFINE:1           93.00   126.00   158.00   187.00   214.00   238.00   260.00   279.00   299.00   327.00   357.00   382.00   405.00   432.00   460.00   487.00   513.00   523.61   541.00   568.00   593.00   616.00   637.00   656.00   675.00   697.00   724.00   751.00   775.00   799.00   825.00   850.00   871.00   895.00   921.00   947.00   970.00   992.00  1017.00  1042.00  1066.00  1093.00  1121.00  1148.00  1173.00  1197.00  1220.00  1242.00  1268.00  1294.00  1320.00  1343.00  1365.00  1389.00  1412.00  1435.00  1461.00  1487.00  1519.00  1539.00
XSEL_REFINE:2           90.00   119.00   154.46   186.00   212.00   234.68   260.88   281.26   304.00   331.00   361.48   387.00   411.00   439.00   468.00   496.00   523.00   536.71   551.00   579.00   606.00   632.00   658.00   677.17   695.28   711.00   739.00   767.00   795.00   820.00   844.00   868.00   890.00   915.00   941.00   967.00   993.00  1016.00  1040.00  1065.00  1089.00  1116.00  1143.00  1170.00  1196.00  1220.00  1244.00  1268.00  1294.00  1320.00  1345.00  1370.00  1392.00  1416.00  1441.00  1463.00  1486.00  1509.00  1531.00  1548.91
</pre>

<h3>Terminology</h3>

<ul>
<li>Trace: An electropherogram trace obtained from electrophoresis experiments.</li>
<li>Data lane (or data row): The set of reactivities obtained from quantifying a single trace or set of reads of a structure mapping experiment for a particular RNA. An experiment usually has multiple data lanes for probing an RNA with different probes or under different conditions.</li>
<li>Annotation: In the RDAT file format, annotations are tags of the form annotationName:annotationValue that can be used to describe an experiment, construct, or data lane.</li>
<li>Standard state: A standard structural mapping experiment for an RNA molecule that can be used as a reference. A standard state experiment usually involves probing the RNA with and without a modifying agent under standard buffer conditions (i.e. without additional chemicals or biological agents).</li>
</ul>

<h3>General section</h3>

<p>The general section contains information about the RDAT specification version
(RDAT_VERSION) used and serves as the root of the document. </p>
<p>The following table describes the attributes that are found in the general section</p>
<table class="spectable">
<tr>
<th>Header</th>
<th>Description</th>
<th>Mandatory?</th>
</tr>
<tr>
<td>RDAT_VERSION</td>
<td>The version of the RDAT file format specification that this RDAT file uses in the format X.XX where X is an integer. 
Current version is 0.32</td>
<td>Yes</td>
</tr>
</table>


<h3>Construct section</h3>

<p>The construct section describes the specific RNA molecule that was probed in the experiment (NAME) and lists
information about the construct, such as nucleotide sequence (SEQUENCE), putative
secondary structure (STRUCTURE), annotation tags (ANNOTATION) and additional free
text comments (COMMENT). Optional fields can specify an integer offset that can
be added to the RNA sequence index to yield a conventional numbering (OFFSET); the
sequence positions to be specified in the file (SEQPOS); and, optionally, the mutation positions of
constructs in the data (MUTPOS), if there are any. An RDAT file may have multiple construct sections,
each one delimited by the NAME attribute of the next construct</p>
<p>The following table describes the attributes that are found in the construct section, in order of appearance</p>
<table class="spectable">
<tr>
<th>Header</th>
<th>Description</th>
<th>Mandatory?</th>
</tr>
<tr>
<td>NAME</td>
<td>The name of the construct in the following format: "RNA system name, Source" where source can be the organism or gene where the system is usually studied (in some cases, one can be more relevant than the other).</pre>. 
If the RNA is of synthetic origin, a source must still be specified.</td>
<td>Yes</td>
</tr>
<tr>
<td>SEQUENCE</td>
<td>The sequence of the RNA molecule studied. The sequence of interest should be in upper case, while auxiliary sequences (e.g. 
primer binding sites or barcodes) are in lower case. Degenerate sequences describing populations of sequences probed can be described
with IUPAC codes (e.g. AGGAAAGGNNRNNAAGAA). This is particularly useful when an experiment compares different sequences that are expected
to fold in the same structure, such as those found in the EteRNA project.</td>
<td>Yes</td>
</tr>
<tr>
<td>STRUCTURE</td>
<td>A string describing the best known secondary structure model of the RNA of interest, in dot bracket notation (e.g. "((....))").
Pseudoknots can be included using "{}" curly braces (e.g "((..{.{))}.}").</td>
<td>Yes</td>
</tr>
<tr>
<td>OFFSET</td>
<td>Sometimes, the sequence studied is a sub-sequence from a bigger sequence (such as the P4-P6 domain of the Tetrahymena ribozyme group I intron) and
the bigger sequence numbering scheme will want to be preserved. The offset then specifies the number such that Pos = OrPos - OFFSET, where Pos is
the position in the subsequence and OrPos is the same position but in the numbering scheme of the bigger sequence. Note that positions are always
considered to be 1-indexed.</td>
<td>No</td>
</tr>
<tr>
<td>SEQPOS</td>
<td>A list of sequence positions and their respective residues that were probed in the experiment. If OFFSET is defined, each position in SEQPOS must have OFFSET already added to it (thus,
a position OrPos defined in SEQPOS refers to position Pos = OrPos - OFFSET in the sequence defined in SEQUENCE.). Positions in SEQPOS are 1-indexed.</td>
<td>No</td>
</tr>
<tr>
<td>ANNOTATION</td>
<td>Annotations for this construct section. See the Annotations section below for more details.</td>
<td>No</td>
</tr>
<tr>
<td>COMMENT</td>
<td>Free text comments for the construct.</td>
<td>No</td>
</tr>
<tr>
<td>XSEL</td>
<td> List of tab-separated integers that specify the place in the traces used for peak calling in the sequence assignment step of the analysis, used globally. This and XSEL_REFINE (see Data Section description below) are mutually exclusive and when provided with both, XSEL will be ignored.</td>
<td>No</td>
</tr>
<tr>
<td>MUTPOS</td>
<td> List of tab-separated values with size of the number of data lanes that indicate which samples have been mutated and where (only single point mutations are accepted in MUTPOS). WT indicates that there is no mutation on the sample.</td>
<td>No</td>
</tr>
</table>
<h3>Data section</h3>
<p>The final required section of the RDAT file includes the mapping data themselves. The data is described through several lists: simple raw data trace or read counts, the positions in the trace that were called as peaks (for electropherograms only), and the quantitated reactivity corresponding to these selections. Errors derived from analysis of multiple replicates can also be included. Each header in this section is followed by a colon and an index (e.g. REACTIVITY:1) in order to identify headers that describe the same data (thus, ANNOTATION_DATA:1 describes annotations for data contained in REACTIVITY:1 and REACTIVITY_ERROR:3 shows errors associated with data in REACTIVITY:3). All attribute values for this section are lists of tab-separated items that may be either annotations (for ANNOTATION_DATA), integers (for XSEL_REFINE and SEQPOS), or real numbers (for REACTIVITY, REACTIVITY_ERROR, TRACE, and READS).</p>
<p>The following table describes the attributes that are found in the construct section, in order of appearance</p>

<table class="spectable">
<tr>
<th>Header</th>
<th>Description</th>
<th>Mandatory?</th>
</tr>
<tr>
<td>ANNOTATION_DATA</td>
<td>Annotations for each data lane. See the Annotations section below for more details.</td>
<td>No</td>
</tr>
<tr>
<td>REACTIVITY</td>
<td>The area peak/likelihood estimate that represents the reactivity at each position, tab-delimited.</td>
<td>Yes</td>
</tr>
<tr>
<td>REACITIVTY_ERROR</td>
<td> If REACTIVITY was derived as a consensus of different replicates, this indicates the standard error between samples used. If only
one experiment was done, this may be some measure of variation between the data in that experiment, e.g. 0.2 of the standard deviation.</td>
<td>No</td>
</tr>
<tr>
<td>XSEL_REFINE</td>
<td> The integers that specify the place in the traces used for peak calling in the sequence assignment step of the analysis, per sample.</td>
<td>No</td>
</tr>
<tr>
<td>SEQPOS</td>
<td>Like SEQPOS of the construct section. However, this would apply only to a specific data lane</p>
<td>No</td>
<tr>
<td>TRACE</td>
<td> The raw electropherogram data, tab-delimited.</td>
<td>No</td>
</tr>
<tr>
<td>READS</td>
<td> The raw read count data (for deep sequencing assays, such as SHAPE-seq), tab-delimited.</td>
<td>No</td>
</tr>
</table>

<h3>Information about ANNOTATION attributes</h3>

<p> The construct and data sections have ANNOTATION or ANNOTATION_DATA attributes that contain annotation tags for the experimental data. Annotations are a central part of the RDAT file format as they give information about the biological and technical context in which the experiment was performed. Annotations are composed of a name and possibly multiple values, separated by a colon (e.g. modifier:SHAPE:DMS) and are listed as tab-separated values after the ANNOTATION or ANNOTATION_DATA headers. Annotations are hierarchical: an annotation in the ANNOTATION attribute in the construct section describes all data contained in that section unless an annotation with the same name in a particular ANNOTATION_DATA attribute overrides it.</p>
<p>The following table describes the annotation syntax for the most common descriptors of an RNA structure mapping experiment.</p>
<p>This annotation list may not fulfill all the needs for experimentalists to accurately describe their data. Contributors are encouraged to contact the RMDB team for inclusions of more annotation types into the specification.</p>

<table class="spectable">
<tr>
<th>Description of annotation</th>
<th>RDAT syntax</th>
<th>SNRNASM terms</th>
</tr>
<tr>
<td>Modified sample using dimethyl sulfate</td>
<td>modifier:DMS</td>
<td>DMS:OBI:001015</td>
</tr>
<tr>
<td>Modified sample using 1-cyclohexyl-(2-morpholinoethyl)carbodiimide metho-p-toluene sulfonate</td>
<td>modifier:CMCT</td>
<td>CMCT:OBI:0001006</td>
</tr>
<tr>
<td>Modified sample using N-methylisatoic anhydride</td>
<td>modifier:NMIA, modifier:SHAPE</td>
<td>NMIA:OBI:0001026</td>
</tr>
<tr>
<td>Modified sample using 1-methyl-7-nitroisatoic anhydride</td>
<td>1M7:in process</td>
<td>modifier:1M7, modifier:SHAPE</td>
</tr>
<tr>
<td>Modified sample using RNase T1</td>
<td>T1:OBI0001030</td>
<td>modifier:RNaseT1</td>
</tr>
<tr>
<td>Modified sample using an enzyme or chemical</td>
<td>See SNRNASM ontology for modifier identifiers</td>
<td>modifier:<i>modifierName</i></td>
</tr>
<tr>
<td>Unmodified sample</td>
<td>modifier:None, modifier:Nomod</td>
<td>-</td>
</tr>
<tr>
<td>Chemical added to samples</td>
<td id="chemical">chemical:<i>chemicalName</i>:<i>concentration</i></td>
<td>-</td>
</tr>
<tr>
<td>Secondary RNA or DNA sequence added to samples</td>
<td>chemical:<i>sequence</i>:<i>concentrarion</i></td>
<td>-</td>
</tr>
<tr>
<td>RNA sequence changed in sample</td>
<td id="sequence">sequence:<i>RNASequence</i></td>
<td>-</td>
</tr>
<tr>
<td>Specify temperature used</td>
<td id="temperature">temperature:<i>temperatureInCelsius</i>C</td>
<td>-</td>
</tr>
<tr>
<td>Specify processing method or algorithm used (such as background substraction or overmodification correction)</td>
<td id="processing">processing:<i>methodName</i></td>
<td>-</td>
</tr>
<tr>
<td>Indicate that the sequence was mutated, from N to M at nucleotide i (1-based sequence indexing). M and N are IUPAC codes for ribonucleoside resiudes.</td>
<td>mutation:NiM</td>
<td>-</td>
</tr>
<tr>
<td>Indicate that the sequence was mutated from NNN to MMM at nucleotide range i to j (1-based indexing). M and N are IUPAC codes for ribonucleoside resiudes.</td>
<td>mutation:NNN(i-j)MMM</td>
<td>-</td>
</tr>
<tr>
<td>Specify experiment type (currently supported: standard state, mutate and map, and titration)</td>
<td id="experimentType">experimentType:StandardState, experimentType:MutateAndMap, or experimentType:Titration</td>
<td>-</td>
</tr>
<tr>
<td>Indicate that a base is modified in some manner at position i. Modifications include 2'-O-methylation, N1-methylation, and N3-methylation.</td>
<td>modification:<i>modification</i>:i</td>
<td>-</td>
</tr>
<tr>
<td>For chemical footprinting experiments, indicate the binding site (positions i to j) for the reverse transcription primer. If this annotation is not present and the modifier is not an enzyme, the binding site will be assumed to be the last 20 nucleotides of the RNA. </td>
<td>primer:(i-j)</td>
<td>-</td>
</tr>
<tr>
<td>Specify the source from where the sequence was extracted. Acceptable sources are pdb:<i>pdbid</i> or even doi:<i>doi</i>. If only part of the sequence was extracted from the indicated source, a position interval (i-j) that corresponds to the extracted subsequence from the original sequence can be specified.</td>
<td>sequenceSource:<i>source</i>:<i>identifier</i> or sequenceSource:<i>source</i>:<i>identifier</i>:(i-j) </td>
<td>-</td>
</tr>

</table>

<h3>Best Practices</h3>
<p> Here we describe the recommended practices for representing RNA mapping data in RDAT format. Although they are not required, these practices make an RDAT file more readable, informative, and easier to parse. </p>
<ul>
<li>Since REACTIVITY_ERROR is usually obtained when comparing reactivities of multiple repeats of the same construct, the corresponding raw traces should not be reported.</li>
<li>When OFFSET has a value other than zero, it probably means that the sequence probed is the subsequence of a bigger RNA. In this case, the annotation sequenceSource should be present to indicate the source of the sequence.</li>
<li>Experiments that are meant to test how well do several sequences fold into a specific structure can be described in RDAT format by setting SEQUENCE to NNNN...NNN (N times the length of the sequences), STRUCTURE to the desired structure, and adding annotations to ANNOTATION_DATA attributes that indicate a change of sequence (e.g. sequence:AAACUCUG).</li>
</ul>
{% endblock %}
